src/test/scala/firrtlTests/SubWordAssignmentTests.scala

```diff
@@ -0,0 +1,342 @@
+    check(src, expected)
+  }
+
+  it should "allow for nested bit indices" in {
+    val src =
+      """circuit m:
+        |  module m:
+        |    output x : UInt<2>
+        |    x[1:0][0][0] <= UInt(1)
+        |    x[1][0][0][0] <= UInt(0)
+        |""".stripMargin
+    val expected =
+      """circuit m :
+        |  module m :
+        |    output x : UInt<2>
+        |    x <= UInt<2>("h1")
+        |""".stripMargin
+    check(src, expected)
+  }
+
+  it should "error on out of bounds hi index" in {
+    val src =
+      """circuit m:
+        |  module m:
+        |    output x : UInt<2>
+        |    x[2:0] <= UInt(1)
+        |""".stripMargin
+    val e = intercept[BitsWidthException] {
+      compile(src)
+    }
+    assert(e.getMessage.contains("High bit 2 in bits operator is larger than input width 2"))
+  }
+
+  it should "error on out of bounds lo index" in {
+    val src =
+      """circuit m:
+        |  module m:
+        |    output x : UInt<2>
+        |    x[1:-1] <= UInt(1)
+        |""".stripMargin
+    val e = intercept[firrtl.passes.CheckHighForm.NegArgException] { compile(src) }
+    assert(e.getMessage.contains("Primop bits argument -1 < 0"))
+  }
+
+  it should "error on hi < lo indices" in {
+    val src =
+      """circuit m:
+        |  module m:
+        |    output x : UInt<2>
+        |    x[0:1] <= UInt(1)
+        |""".stripMargin
+    val e = intercept[firrtl.passes.CheckHighForm.LsbLargerThanMsbException] { compile(src) }
+    assert(e.getMessage.contains("Primop bits lsb 1 > 0"))
+  }
+
+  it should "error when assigning an input" in {
+    val src =
+      """circuit m:
+        |  module m:
+        |    input x : UInt<2>
+        |    x[0] <= UInt(1)
+        |""".stripMargin
+    val e = intercept[firrtl.passes.CheckFlows.WrongFlow] { compile(src) }
+    assert(e.getMessage.contains("is used as a SinkFlow but can only be used as a SourceFlow"))
+  }
+
+  it should "error when assigning an input inside a bundle" in {
+    val src =
+      """circuit m:
+        |  module m:
+        |    output io : { flip x : UInt<2> }
+        |    io.x[0] <= UInt(1)
+        |""".stripMargin
+    val e = intercept[firrtl.passes.CheckFlows.WrongFlow] { compile(src) }
+    assert(e.getMessage.contains("is used as a SinkFlow but can only be used as a SourceFlow"))
+  }
+
+  // TODO: should we support these kinds of word-level, but not bit-level loops?
+  it should "allow signals to be connected trough bit slices" ignore {
+    val src =
+      """circuit m:
+        |  module m:
+        |    input y : UInt<1>
+        |    output x : UInt<1>
+        |    wire tmp : UInt<4>
+        |    x <= not(tmp[3])
+        |    tmp[3:1] <= not(tmp[2:0])
+        |    tmp[0] <= y
+        |""".stripMargin
+    val expected =
+      """circuit m :
+        |  module m :
+        |    input y : UInt<1>
+        |    output x : UInt<1>
+        |    x <= y
+        |""".stripMargin
+    check(src, expected)
+  }
```

Comment:
Might also want a test for subword-assigning an SInt:

```
circuit test :
  module test :
    input c : UInt<2>
    output x : SInt<2>

    x[0] <= c[0]
    x[1] <= c[1]
```

Note that since the type of `x[...]` is always `UInt`, it must be connected to a `UInt` even though `x` is an `SInt`.

Reply:
Yes. Thanks, I need to add this test (and fix the implementation to support it)